message.go

```diff
@@ -0,0 +1,670 @@
+	ProxyURI:      optionDef{valueFormat: valueString, minLen: 1, maxLen: 1034},
+	ProxyScheme:   optionDef{valueFormat: valueString, minLen: 1, maxLen: 255},
+	Size1:         optionDef{valueFormat: valueUint, minLen: 0, maxLen: 4},
+}
+
+// MediaType specifies the content type of a message.
+type MediaType uint16
+
+// Content types.
+const (
+	TextPlain         MediaType = 0     // text/plain;charset=utf-8
+	AppCoseEncrypt0   MediaType = 16    // application/cose; cose-type="cose-encrypt0" (RFC 8152)
+	AppCoseMac0       MediaType = 17    // application/cose; cose-type="cose-mac0" (RFC 8152)
+	AppCoseSign1      MediaType = 18    // application/cose; cose-type="cose-sign1" (RFC 8152)
+	AppLinkFormat     MediaType = 40    // application/link-format
+	AppXML            MediaType = 41    // application/xml
+	AppOctets         MediaType = 42    // application/octet-stream
+	AppExi            MediaType = 47    // application/exi
+	AppJSON           MediaType = 50    // application/json
+	AppJsonPatch      MediaType = 51    //application/json-patch+json (RFC6902)
+	AppJsonMergePatch MediaType = 52    //application/merge-patch+json (RFC7396)
+	AppCBOR           MediaType = 60    //application/cbor (RFC 7049)
+	AppCWT            MediaType = 61    //application/cwt
+	AppCoseEncrypt    MediaType = 96    //application/cose; cose-type="cose-encrypt" (RFC 8152)
+	AppCoseMac        MediaType = 97    //application/cose; cose-type="cose-mac" (RFC 8152)
+	AppCoseSign       MediaType = 98    //application/cose; cose-type="cose-sign" (RFC 8152)
+	AppCoseKey        MediaType = 101   //application/cose-key (RFC 8152)
+	AppCoseKeySet     MediaType = 102   //application/cose-key-set (RFC 8152)
+	AppCoapGroup      MediaType = 256   //coap-group+json (RFC 7390)
+	AppOcfCbor        MediaType = 10000 //application/vnd.ocf+cbor
+	AppLwm2mTLV       MediaType = 11542 //application/vnd.oma.lwm2m+tlv
+	AppLwm2mJSON      MediaType = 11543 //application/vnd.oma.lwm2m+json
+)
+
+type option struct {
+	ID    OptionID
+	Value interface{}
+}
+
+func encodeInt(v uint32) []byte {
+	switch {
+	case v == 0:
+		return nil
+	case v < 256:
+		return []byte{byte(v)}
+	case v < 65536:
+		rv := []byte{0, 0}
+		binary.BigEndian.PutUint16(rv, uint16(v))
+		return rv
+	case v < 16777216:
+		rv := []byte{0, 0, 0, 0}
+		binary.BigEndian.PutUint32(rv, uint32(v))
+		return rv[1:]
+	default:
+		rv := []byte{0, 0, 0, 0}
+		binary.BigEndian.PutUint32(rv, uint32(v))
+		return rv
+	}
+}
+
+func decodeInt(b []byte) uint32 {
+	tmp := []byte{0, 0, 0, 0}
+	copy(tmp[4-len(b):], b)
+	return binary.BigEndian.Uint32(tmp)
+}
+
+func (o option) toBytes() []byte {
+	var v uint32
+
+	switch i := o.Value.(type) {
+	case string:
+		return []byte(i)
+	case []byte:
+		return i
+	case MediaType:
+		v = uint32(i)
+	case int:
+		v = uint32(i)
+	case int32:
+		v = uint32(i)
+	case uint:
+		v = uint32(i)
+	case uint32:
+		v = i
+	default:
+		panic(fmt.Errorf("invalid type for option %x: %T (%v)",
```

Comment:
if this is not really needed consider returning error instead of panicing

Reply:
Now it returns error and function upper just log fatal error for that option.